src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -244,54 +244,20 @@ public override void WriteCSharpExpressionAttributeValue(CodeRenderingContext co
                 return;
             }
 
-            var firstChild = node.Children[0];
-            if (firstChild.Source != null)
+            context.CodeWriter.WriteStartAssignment(DesignTimeVariable);
+            for (var i = 0; i < node.Children.Count; i++)
             {
-                using (context.CodeWriter.BuildLinePragma(firstChild.Source.Value, context))
+                if (node.Children[i] is IntermediateToken token && token.IsCSharp)
```

Comment:
Don't we need the padding here? I see that it was removed

Reply:
`WriteCSharpToken` takes care of adding the padding.